pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -361,19 +369,20 @@ func (vm *windows) waitForHNSNetworks() error {
 	return errors.Wrap(err, "timeout waiting for OVN overlay HNS networks")
 }
 
-// waitForHybridOverlayToRun waits for the hybrid-overlay-node.exe to run until the timeout is reached
+// waitForHybridOverlayToRun waits for the hybrid-overlay-node service to reach RUNNING state
+// until the timeout is reached
 func (vm *windows) waitForHybridOverlayToRun() error {
 	var err error
 	for retries := 0; retries < retry.Count; retries++ {
-		_, err = vm.Run("Get-Process -Name \""+HybridOverlayProcess+"\"", true)
-		if err == nil {
+		out, _ := vm.Run("sc.exe query \""+hybridOverlayServiceName+"\" | findstr RUNNING", true)
```

Comment:
What would happen if the service goes momentarily to running state and we return as success? Would it cause any problems or would it be caught as the HNS networks won't be created?

My main question is should we return when we see that our service goes to `RUNNING` or should we wait for few minutes and see `no failure`?

Reply:
I am not sure if there can be a scenario like that AFAI can see from the code at: https://github.com/openshift/ovn-kubernetes/blob/2640a627e0a0c58d1d3fb71046f0bbf527522858/go-controller/hybrid-overlay/pkg/windows/service/service.go#L51, if there is an error service would not go to running state it will be returned before that and the only scenarios for service to go to start_pending after running are when the ctx is done or the service is stopped. To further add yes the HNSNetworks wont get created if there is an error after the service has been in running state. This function will just help us to debug more in the lines of whether the error was network related or service related. IMO we should be adding wait if its required.